### DISCUSSION TITLE
Install suport debian 11

### DISCUSSION THREAD

**Discussion by arsenieciprian** [hnimminh/libresbc#72] [General]
Hi.
Can u please help me to install libresbc in to my debain 11 server.
skype id is arsenieciprian

**Comment by hnimminh** [comment]
@arsenieciprian have you follow installation guide on https://github.com/hnimminh/libresbc/wiki/Installation ?

**Reply by arsenieciprian** [reply]
I m not verry familiar with ansible. i wnat to install directley on debian 11

**Reply by hnimminh** [reply]
let try it.

**Reply by jj1682000** [reply]
Sorry, I have follow installation guide on https://github.com/hnimminh/libresbc/wiki/Installation.
but it failed:

$ ansible-playbook playbooks/deployment.yml -i inventories/production -l "sbc01" -t "platform,libre,nginx,captagent" 
PLAY [LibreSBC Deployment] ******************************************************************************************************

TASK [Gathering Facts] **********************************************************************************************************
ok: [sbc01]

TASK [platform : Disable SELinux] ***********************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ModuleNotFoundError: No module named 'selinux'
fatal: [sbc01]: FAILED! => {"changed": false, "msg": "Failed to import the required Python library (libselinux-python) on sbc01's Python /usr/bin/python3. Please read the module documentation and install it in the appropriate location. If the required library is installed, but Ansible is using the wrong Python interpreter, please consult the documentation on ansible_python_interpreter"}
...ignoring

TASK [platform : Update apt-cache] **********************************************************************************************
ok: [sbc01]

TASK [platform : Update all packages to their latest version] *******************************************************************
ok: [sbc01]

TASK [platform : Upgrade the OS (apt-get dist-upgrade)] *************************************************************************
ok: [sbc01]

TASK [platform : Set timezone to UTC] *******************************************************************************************
ok: [sbc01]

TASK [platform : Change hostname sbc01 to sbc01] ********************************************************************************
ok: [sbc01]

TASK [platform : set hostname - workaround] *************************************************************************************
ok: [sbc01]

TASK [platform : Remove unused packages] ****************************************************************************************
ok: [sbc01]

TASK [platform : Install essential packages] ************************************************************************************
fatal: [sbc01]: FAILED! => {"changed": false, "msg": "No package matching 'g++-8' is available"}

PLAY RECAP **********************************************************************************************************************
sbc01                      : ok=9    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=1

**Comment by WEBudoGT** [comment]
Hi
I'm also unfamiliar with ansible and want to install manually.
I executed the installation.debian.sh commands by hand because it was failing due to some typos and other stuff, and now I finished running everything, but I can't see where/how to run/compile libresbc.
What shouls I do after git cloning it?

Thanks!

**Reply by WEBudoGT** [reply]
forget it... I got one instance running using ansible (it's awesome!)

**Reply by hnimminh** [reply]
awesome! using ansible is recommendation.

**Reply by jchinyou** [reply]
Hey @WEBudoGT 

Would you be able to share how you got around the deprecated g++-8 in Debian 11 with Ansible? 

This is my first install of Libresbc and I'm blindly hacking my way through the config right now to try and see if I can get Ansible to install on Deb 11. 

******************************************************************************************************************************
2022-12-24 01:11:47,149 p=1035 u=ec2-user n=ansible | fatal: [libresbc-primary]: FAILED! => {"changed": false, "msg": "No package matching 'g++-8' is available"}
2022-12-24 01:11:47,152 p=1035 u=ec2-user n=ansible | PLAY RECAP ****************************************************************************************************************************************************************

**Reply by WEBudoGT** [reply]
hello @jchinyou 
I couldn't get it running using Ansible on Debian 11. Got to go back to Debian 10 and installed flawlesly.

**Reply by jchinyou** [reply]
Thanks @WEBudoGT !

**Comment by arsenieciprian** [comment]
Can u share some screenshuts with web interface ?Trimis de pe iPhone‑ul meuPe 7 nov. 2022, la 19:46, Alex Mejia ***@***.***> a scris:﻿
forget it... I got one instance running using ansible (it's awesome!)

**Comment by jchinyou** [comment]
Hi @hnimminh 

Wouldn’t suppose you may have scripts for Deb 11 floating around in a Dev Branch somewhere just yet? 

If not no worries, I just happen to have some environment challenges I’m working against.

**Reply by hnimminh** [reply]
ah, It's not available.. can you help to create a feature request for this - thanks

**Comment by ciscomonkey** [comment]
I created PR #97 with the changes to allow install on 10 or 11.

**Comment by hectoryanezq** [comment]
buenas lguien que me ayude como hacerlo, segui los pasos en nsible pero no pude lograr instalarlo, tambien por docker pero no veo la interfaz WEBUI
![image](https://github.com/hnimminh/libresbc/assets/93660809/f0a16eea-2249-4100-99f3-7c54df453a0e)

AYUDAA plis

**Reply by hectoryanezq** [reply]
con un ejemplo de ansible.cfg  y host.yml creo que puedo logralo

se los agradeceria muchisimo

**Reply by ciscomonkey** [reply]
The relevant section of the host.yml file should be this (based on the error output):

```yaml
  hosts:
    <machine-name>:
      ansible_host: <libresbc-machine-ip>
      nodeid: <unique-nodeid>
```

Which should be updated to look something like this:

```yaml
  hosts:
    libresbc1:
      ansible_host: 1.1.1.1
      nodeid: libresbc1
```

I noticed that it looks like you might have a dash in the hostname in your error-output above. If so, just be aware that will cause you some issues if you use that as the nodeid - see [#83](https://github.com/hnimminh/libresbc/issues/83).

Then in your ansible.cfg you'd want to update those settings under ```[defaults]``` to match your environment - here's the ones I modified in mine:

```
[defaults]
private_key_file    = ~/.ssh/yourkeygoeshere
remote_user         = root
inventory           = inventories/do 
#vault_password_file = ~/.ssh/libresbc.ansible.vault
```

**Reply by hectoryanezq** [reply]
agradecido por la ayudar, funciono pero hasta cierto punto, me aparecio un error enorme lo cual lo he echo al menos 4 veces mas y no funciona
![image](https://github.com/hnimminh/libresbc/assets/93660809/c1ab6385-976c-422f-b6c4-3cd4ab66bd41)

![image](https://github.com/hnimminh/libresbc/assets/93660809/218a96ea-d284-4695-b523-fbb5a38d5148)

**Reply by hnimminh** [reply]
What is your debian version? and what is ansible command you ran? 
can upload the ansible log to here?

**Reply by hnimminh** [reply]
WebUI is not include in docker/ansible.
it is an add-on, you can found here on how to make use it https://github.com/hnimminh/libresbc/discussions/120

**Reply by hectoryanezq** [reply]
![image](https://github.com/hnimminh/libresbc/assets/93660809/33e55546-cda6-4090-a875-44d6d6a36b3d)
 ese es el cliente,donde quiero instalar libresbc

ansible-playbook playbooks/deployment.yml -i inventories/production/hosts.yml -l "libresbc" -t "platform,libre,nginx,captagent" -e "signalwire_pat_token=MITOKEN"  --ask-vault-password

este es el comando que use, incluso, antes de lanzar el comando, me conecte al cliente y lance apt-get build-dep freeswitch, asi pude seguir con el ansible, sino me seguia saliendo el error de la foto anterior,

actualmente denuevo quede con el siguiente error
![image](https://github.com/hnimminh/libresbc/assets/93660809/9bed8fbd-8ac9-437a-9f21-8ac21f6545b1)


gracias por ayudarme

**Reply by hectoryanezq** [reply]
version 10.13 debian

**Reply by hectoryanezq** [reply]
TASK [libre : Copy liberator role source to target host] ************************************************************************************************************************************************************************************
[DEPRECATION WARNING]: The connection's stdin object is deprecated. Call display.prompt_until(msg) instead. This feature will be removed in version 2.19. Deprecation warnings can be disabled by setting deprecation_warnings=False in
ansible.cfg.
fatal: [libresbc]: FAILED! => {"changed": false, "cmd": "/usr/bin/rsync --delay-updates -F --compress --archive --rsh='/usr/bin/ssh -S none -i /root/.ssh/id_rsa -o Port=22 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null' --rsync-path='sudo -u root rsync' --exclude=*.pyc --out-format='<<CHANGED>>%i %n%L' /root/libresbc/liberator/ root@192.168.103.249:/opt/libresbc/v0.7.1.b/liberator/", "msg": "Warning: Permanently added '192.168.103.249' (ECDSA) to the list of known hosts.\r\nbash: sudo: orden no encontrada\nrsync: connection unexpectedly closed (0 bytes received so far) [sender]\nrsync error: error in rsync protocol data stream (code 12) at io.c(228) [sender=3.2.3]\n", "rc": 12}

**Reply by hectoryanezq** [reply]
logre poder instalarlo pero la interfas WEBUI no se logra ver

![image](https://github.com/hnimminh/libresbc/assets/93660809/f4953947-7dbe-4878-a06b-b7d4f3585699)


![image](https://github.com/hnimminh/libresbc/assets/93660809/ea00355a-48ed-477f-bf1f-07c3bc49af67)
, alguien que lo pueda mostrar con interfas UI y como lo hizo para levantarla

plisss

**Reply by hnimminh** [reply]
First, it seem you set incorrect `debian_codename`, debian 10 is **buster**, and 11 is **bullseye**
Second, WebUI can be run on your personal computer. If you run on the server where libresbc installed. You need allow port in in the firewall.

**Reply by hectoryanezq** [reply]
era porque no tenia instalado sudo error mio Xd, el siguiente punto

a que te refieres con firewall, mi firewall de red, o el firewall del Sistema Operativo

porque si puedo acceder a la API
![image](https://github.com/hnimminh/libresbc/assets/93660809/26f6f9b0-0202-4ad8-a00b-69200e414652)
 si fuese el firewall de mi red, no veria esto

Gracias por la respuesta

**Reply by hnimminh** [reply]
Where did you run libreui? i mean `go run ..` command

**Reply by hectoryanezq** [reply]
![image](https://github.com/hnimminh/libresbc/assets/93660809/ced6221b-00f0-4981-a085-64e4343d6366)
lo levante de esta forma

**Reply by hnimminh** [reply]
I guess (from the above screenshot) you run it on the same machine as libresbc, if so, check your firewall.
reason because the UI listen on 8088
